pkg/controller/migmigration/task.go

```diff
@@ -39,52 +39,82 @@ const (
 	Completed                     = "Completed"
 )
 
-// Steps
-var Step = []string{
-	Created,
-	Started,
-	Prepare,
-	EnsureInitialBackup,
-	InitialBackupCreated,
-	InitialBackupFailed,
-	AnnotateResources,
-	EnsureStagePods,
-	StagePodsCreated,
-	RestartRestic,
-	ResticRestarted,
-	QuiesceApplications,
-	EnsureQuiesced,
-	EnsureStageBackup,
-	StageBackupCreated,
-	StageBackupFailed,
-	EnsureStageBackupReplicated,
-	EnsureStageRestore,
-	StageRestoreCreated,
-	StageRestoreFailed,
-	EnsureStagePodsDeleted,
-	EnsureAnnotationsDeleted,
-	EnsureInitialBackupReplicated,
-	EnsureFinalRestore,
-	FinalRestoreCreated,
-	FinalRestoreFailed,
-	Completed,
+// Flags
+const (
+	QUIESCE = 0x01 // Only when QuiescePods (true).
+	PVS     = 0x02 // Only when PVs migrated.
+)
+
+type Itinerary []Step
+
+var StageItinerary = Itinerary{
+	{phase: Created},
+	{phase: Started},
+	{phase: Prepare},
+	{phase: AnnotateResources},
+	{phase: EnsureStagePods, flags: PVS},
+	{phase: StagePodsCreated, flags: PVS},
+	{phase: RestartRestic, flags: PVS},
+	{phase: ResticRestarted, flags: PVS},
+	{phase: QuiesceApplications, flags: QUIESCE},
+	{phase: EnsureQuiesced, flags: QUIESCE},
+	{phase: EnsureStageBackup},
+	{phase: StageBackupCreated},
+	{phase: EnsureStageBackupReplicated},
+	{phase: EnsureStageRestore},
+	{phase: StageRestoreCreated},
+	{phase: EnsureStagePodsDeleted},
+	{phase: EnsureAnnotationsDeleted},
+	{phase: Completed},
 }
 
-// End phases.
-var EndPhase = map[string]bool{
-	InitialBackupFailed: true,
-	StageBackupFailed:   true,
-	FinalRestoreFailed:  true,
-	StageRestoreFailed:  true,
-	Completed:           true,
+var FinalItinerary = Itinerary{
+	{phase: Created},
+	{phase: Started},
+	{phase: Prepare},
+	{phase: EnsureInitialBackup},
+	{phase: InitialBackupCreated},
+	{phase: AnnotateResources},
+	{phase: EnsureStagePods, flags: PVS},
+	{phase: StagePodsCreated, flags: PVS},
+	{phase: RestartRestic, flags: PVS},
+	{phase: ResticRestarted, flags: PVS},
+	{phase: QuiesceApplications, flags: QUIESCE},
+	{phase: EnsureQuiesced, flags: QUIESCE},
+	{phase: EnsureStageBackup},
```

Comment:
`flags: PVS` for EnsureStageBackup through EnsureAnnotationsDeleted

Reply:
Ah, right.

Reply:
Fixed.